### PR TITLE
(DOCSP-42170) BAAS release notes for 20240807

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-42170

I only included a few bug fixes. The rest of the items seem to be internal implementation details. Though maybe we want to mention the CLI not showing some graphql/hosting fields in output? I'm split on whether that info should be included here or not.

### Staging Links

<li><a href=https://deploy-preview-823--app-services.netlify.app/release-notes/backend>release-notes/backend</a></li>

- [Atlas App Services Changelog](https://deploy-preview-823--app-services.netlify.app/release-notes/backend/)

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Did you tag pages appropriately?
  - genre
  - programming_language
  - meta.keywords
  - meta.description
- [x] Describe your PR's changes in the Release Notes section
- [ ] Create a Jira ticket for related docs-realm work, if any

### Release Notes

 - Update external changelog with release notes for several bug fixes.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/release-notes/backend.txt [hunk 1]
```diff
@@ -13,6 +13,15 @@ Atlas App Services Changelog
 2024 Releases
 -------------
 
+.. _backend_20240807:
+
+07 August 2024 Release
+~~~~~~~~~~~~~~~~~~~~~~
+
+- Fixed bug where some destructive changes in push-pull were not being handled properly
```
Comment (dacharyc) [inline source/release-notes/backend.txt:21]: These seem to be all related to Sync, so maybe we want to put them under a "Device Sync" header? It looks like we have been inconsistent about this in the past - sometimes we have listed the specific part of the product that the release notes pertain to (i.e. Edge Server, CLI), but other times we don't.

This is probably a good high-level thing to consider as we're pondering our release notes process.

File: source/release-notes/backend.txt [hunk 2]
```diff
@@ -13,6 +13,15 @@ Atlas App Services Changelog
...
+- Fixed bug where unlinking a datasource with Sync enabled would crash the UI
```
Comment (dacharyc) [inline source/release-notes/backend.txt:22]: When we talk about data sources in the docs, we tend not to concatenate the term. i.e. in the ToC, we have: "Connect to MongoDB Data Sources".

```suggestion
- Fixed bug where unlinking a data source with Sync enabled would crash the UI
```

**Review by dacharyc — APPROVED** [review]
No blocking changes, really - just a Q and a small suggestion.

File: source/release-notes/backend.txt (hunk 1 above)
Comment (krollins-mdb) [reply source/release-notes/backend.txt:21]: Yep, makes sense!

**Pull request merged**